packages/fuel-gauge/src/contract.test.ts

```diff
@@ -691,6 +691,83 @@ describe('Contract', () => {
     expect(finalBalance).toBe(initialBalance + amountToContract.toNumber());
   });
 
+  it('should transferToContract with a large amount of assets', async () => {
+    using launched = await launchTestNode({
+      contractsConfigs,
+      walletsConfig: {
+        amountPerCoin: 2 ** 62,
+      },
+    });
+    const {
+      provider,
+      wallets: [wallet],
+      contracts: [contract],
+    } = launched;
+
+    const initialBalance = new BN(await contract.getBalance(provider.getBaseAssetId())).toNumber();
+    const amountToContract = bn(2).pow(62); // Very big number
+
+    const tx = await wallet.transferToContract(
+      contract.id,
+      amountToContract,
+      provider.getBaseAssetId()
+    );
+
+    await tx.waitForResult();
+
+    const finalBalance = new BN(await contract.getBalance(provider.getBaseAssetId())).toString();
+    expect(finalBalance).toBe(amountToContract.add(initialBalance).toString());
+  });
+
+  it('should batch transfer with a large amount of assets', async () => {
+    using launched = await launchTestNode({
+      contractsConfigs,
+      walletsConfig: {
+        amountPerCoin: 2 ** 62,
+      },
+    });
+    const {
+      provider,
+      wallets: [wallet],
+      contracts: [contract],
+    } = launched;
+
+    const baseAssetId = provider.getBaseAssetId();
+    const contractTransferParams: ContractTransferParams[] = [
+      {
+        contractId: contract.id,
+        amount: bn(2).pow(50),
+        assetId: baseAssetId,
+      },
+      {
+        contractId: contract.id,
+        amount: bn(2).pow(10),
+        assetId: baseAssetId,
+      },
+    ];
+
+    const tx = await wallet.batchTransferToContracts(contractTransferParams);
+
+    const { receipts } = await tx.waitForResult();
+
+    const transferReceipts = receipts.filter(
+      ({ type }) => type === ReceiptType.Transfer
+    ) as ReceiptTransfer[];
+
+    expect(transferReceipts.length).toBe(contractTransferParams.length);
+
+    contractTransferParams.forEach(({ amount, contractId, assetId = baseAssetId }) => {
+      const foundReceipt = transferReceipts.find(
+        (r) =>
+          r.amount.eq(amount) &&
+          r.to.toLowerCase() === contractId.toString().toLowerCase() &&
+          r.assetId === assetId
+      );
+
+      expect(foundReceipt).toBeDefined();
```

Comment:
How come you went for the reciept check here over the contract balance check?

Reply:
If the receipt was produced, the transfer happened. There is no need to validate the contract balance.

Reply:
Understood, was just a different assertion to the previous test, so was just wondering.